src/array.rs

```diff
@@ -123,16 +136,19 @@ impl<T: Miniconf, const N: usize> Miniconf for Array<T, N> {
         item.set_by_key(keys, de).increment()
     }
 
-    fn get_by_key<K, S>(&self, mut keys: K, ser: S) -> Result<S::Error>
+    fn traverse_by_key<K, F, E>(mut keys: K, mut func: F) -> Result<usize, Error<E>>
     where
         K: Iterator,
         K::Item: Key,
-        S: serde::Serializer,
+        F: FnMut(usize, &str) -> Result<(), E>,
     {
         let key = keys.next().ok_or(Error::TooShort(0))?;
         let index = key.find::<Self>().ok_or(Error::NotFound(1))?;
-        let item = self.0.get(index).ok_or(Error::NotFound(1))?;
-        item.get_by_key(keys, ser).increment()
+        if index >= N {
+            return Err(Error::NotFound(1));
+        }
+        func(index, itoa::Buffer::new().format(index))?;
+        T::traverse_by_key(keys, func).increment()
```

Comment:
This kind of recursion might cause stack usage to get _really high_ if you have a large array of elements.

Reply:
It only recurses depth, not breadth.
The stack usage and recursion pattern is the same as before.

Reply:
Ah you're right. Then I'm not too concerned